pkg/apis/internal.admin.acorn.io/v1/quotarequests_test.go

```diff
@@ -0,0 +1,492 @@
+				BaseResources: BaseResources{},
+			},
+		},
+		{
+			name: "remove persistent counts with all",
+			current: QuotaRequestResources{
+				BaseResources: BaseResources{
+					VolumeStorage: resource.MustParse("1Mi"),
+				},
+				Secrets: 1,
+			},
+			incoming: QuotaRequestResources{
+				BaseResources: BaseResources{
+					VolumeStorage: resource.MustParse("1Mi"),
+				},
+				Secrets: 1,
+			},
+			all: true,
+			expected: QuotaRequestResources{
+				BaseResources: BaseResources{},
+			},
+		},
+		{
+			name: "does not remove persistent counts without all",
+			current: QuotaRequestResources{
+				BaseResources: BaseResources{
+					VolumeStorage: resource.MustParse("1Mi"),
+				},
+				Secrets: 1,
+			},
+			incoming: QuotaRequestResources{
+				BaseResources: BaseResources{
+					VolumeStorage: resource.MustParse("1Mi"),
+				},
+				Secrets: 1,
+			},
+			expected: QuotaRequestResources{
+				BaseResources: BaseResources{
+					VolumeStorage: resource.MustParse("1Mi"),
+				},
+				Secrets: 1,
+			},
+		},
+		{
+			name: "remove where current has a resource specified with unlimited",
+			current: QuotaRequestResources{
+				BaseResources: BaseResources{
+					Apps:   Unlimited,
+					Memory: UnlimitedQuantity(),
+				},
+				Secrets: Unlimited,
+			},
+			incoming: QuotaRequestResources{
+				BaseResources: BaseResources{
+					Apps:   1,
+					Memory: resource.MustParse("1Mi"),
+				},
+				Secrets: 1,
+			},
+			expected: QuotaRequestResources{
+				BaseResources: BaseResources{
+					Apps:   Unlimited,
+					Memory: UnlimitedQuantity(),
+				},
+				Secrets: Unlimited,
+			},
+		},
+		{
+			name: "remove where incoming has a resource specified with unlimited",
+			current: QuotaRequestResources{
+				BaseResources: BaseResources{
+					Apps:   1,
+					Memory: resource.MustParse("1Mi"),
+				},
+				Secrets: 1,
+			},
+			incoming: QuotaRequestResources{
+				BaseResources: BaseResources{
+					Apps:   Unlimited,
+					Memory: UnlimitedQuantity(),
+				},
+				Secrets: Unlimited,
+			},
+			expected: QuotaRequestResources{
+				BaseResources: BaseResources{
+					Apps:   1,
+					Memory: resource.MustParse("1Mi"),
+				},
+				Secrets: 1,
+			},
+		},
+		{
+			name: "remove where current and incoming have a resource specified with unlimited",
+			current: QuotaRequestResources{
+				BaseResources: BaseResources{
+					Apps:   Unlimited,
+					Memory: UnlimitedQuantity(),
+				},
+			},
+			incoming: QuotaRequestResources{
+				BaseResources: BaseResources{
+					Apps:   Unlimited,
+					Memory: UnlimitedQuantity(),
+				},
+			},
+			expected: QuotaRequestResources{
+				BaseResources: BaseResources{
+					Apps:   Unlimited,
+					Memory: UnlimitedQuantity(),
+				},
+			},
+		},
+	}
+	// Run the test cases
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			tc.current.Remove(tc.incoming, tc.all)
+			assert.True(t, tc.current.Equals(tc.expected))
+		})
+	}
+}
+func TestEquals(t *testing.T) {
```

Comment:
suggestion: add two more test cases: one where the only thing unequal between `current` and `incoming` is `Secrets`, and one where the only thing unequal between `current` and `incoming` is something inside of `BaseResources`.

This would help make sure that both the logic that checks the secrets is correct, as well as the logic that calls the equal function on `BaseResources`.

Reply:
Good call. I added some similar cases to fits for similar reasons.